src/Console/LintCommand.php

```diff
@@ -2,96 +2,100 @@
             ->addOption('reporter', 'r', InputOption::VALUE_REQUIRED, 'The reporter to use.', 'console')
             ->addOption('ruleset', null, InputOption::VALUE_REQUIRED, 'Ruleset class to use', Official::class)
+            ->addOption('config', null, InputOption::VALUE_REQUIRED, 'Config file to use', null)
         ;
     }
 
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int|null
+     * @throws \Exception
+     */
     public function execute(InputInterface $input, OutputInterface $output)
     {
         $container = $this->getContainer();
-        $limit = $this->getSeverityLimit($input);
-
-        $paths = $input->getArgument('paths');
-        $exclude = $input->getOption('exclude');
-
-        $files = [];
-        foreach ($paths as $path) {
-            if (is_file($path)) {
-                $files[] = new \SplFileInfo($path);
-            } else {
-                $finder = new Finder();
-                $found = iterator_to_array($finder->in($path)->exclude($exclude)->name('*.twig'));
-                if (!empty($found)) {
-                    $files = array_merge($files, $found);
-                }
-            }
-        }
+        $twig = $container['twig'];
+        $validator = $container['validator'];
+        $resolver = new ConfigurationResolver($container, getcwd(), [
+            'path' => $input->getArgument('path'),
+            'path-mode' => $input->getOption('path-mode'),
+            'reporter-service-name' => $input->getOption('reporter'),
+            'exclude' => $input->getOption('exclude'),
+            'severity' => $input->getOption('severity'),
+            'ruleset-class-name' => $input->getOption('ruleset'),
+            'config' => $input->getOption('config'),
+        ]);
+        $severityLimit = $resolver->getSeverityLimit();
+        $finder = $resolver->getFinder();
+        $reporter = $resolver->getReporter();
 
         $violations = [];
 
-        $ruleset = $input->getOption('ruleset');
+        $finderSanitizer = new FinderSanitizer();
+        $files = $finderSanitizer->sanitize($finder);
+//        $files = $finder;
 
-        if (!class_exists($ruleset)) {
-            throw new \InvalidArgumentException(sprintf('Ruleset class %s does not exist', $ruleset));
-        }
+        // TODO: should be refactored to a Runner class
+        foreach ($files as $file) {
+            /** @var SmartFileInfo $file */
+//            var_dump(
+//                'getRelativeFilePath',
+//                $file->getRelativeFilePath(),
+//                $file->getRelativePathname(),
+//                ''
+//                'getRelativeFilePathFromCwd',
+//                $file->getRelativeFilePathFromCwd(),
+//                'getRelativePathname',
+//                $file->getRelativePathname(),
+//                'getRelativeFilePathFromDirectory',
+//                $file->getRelativeFilePathFromDirectory(getcwd())
+//            );
```

Comment:
Code which is commented out shouldn't be part of the PR. If it is required for refactoring one should restore it using the version control system at hand.

Reply:
thats of course clear. it's a RFC PR